langchain4j-core/src/main/java/dev/langchain4j/data/document/Metadata.java

```diff
@@ -124,6 +124,23 @@ public String getString(String key) {
             "It cannot be returned as a String.", key, value, value.getClass().getName());
     }
 
+    public Boolean getBoolean(String key) {
```

Comment:
Why is this required? It is not that easy, we need to:
- make sure most stores support this data type
- add support for this data type in `Filter`
- add more tests to `EmbeddingStoreWithFilteringIT`
- etc

Reply:
I'll revert this